src/pages/signin/ResendValidationForm.js

```diff
@@ -101,6 +101,8 @@ class ResendValidationForm extends React.Component {
             });
         } else if (isOldUnvalidatedAccount) {
             message = this.props.translate('resendValidationForm.unvalidatedAccount');
+        } else if (this.props.account.validateCodeExpired) {
+            message = this.props.translate('resendValidationForm.validationCodeFailedMessage');
```

Comment:
By swapping these two statements we don't need `isOldUnvalidatedAccount` condition to depend on `validateCodeExpired`.

```suggestion
        } else if (this.props.account.validateCodeExpired) {
            message = this.props.translate('resendValidationForm.validationCodeFailedMessage');
        } else if (isOldUnvalidatedAccount) {
            message = this.props.translate('resendValidationForm.unvalidatedAccount');
```

Reply:
Swapping the condition was the first solution I came up with. I was a bit concerned about the value of `isOldUnvalidatedAccount` in the code and hence added `validateCodeExpired ` as well.

Reply:
@marcaaron Can you please check now?

Reply:
>Swapping the condition was the first solution I came up with. I was a bit concerned about the value of isOldUnvalidatedAccount in the code and hence added validateCodeExpired as well.

This is why we suggest to ask questions.

I would recommend in future reviews to:

1. Confirm what needs to be done with your reviewer to be sure you understand
2. Ask questions about things you don't understand
3. Make the changes and request a review after you are sure there is a mutual understanding

>And I thought it is too simple to be true

I understand, but code is not supposed to be magic. There is no "too simple to be true" there's only confidence in the logic or lack of knowledge about how it works. If you don't understand something then please ask for help in Slack or the review.

Reply:
Thanks for the detailed feedback! Really appreciate you spending so much time explaining and being patient on this one.